lib/traject/macros/custom.rb

```diff
@@ -205,14 +205,26 @@ def self.includes_oclc_indicators?(sf_a)
           sf_a.include?('OCLC')
       end
 
-      # Extract ht_id
-      def extract_ht_id
+      # Extract ht_bib_key
+      def extract_ht_bib_key
         lambda do |_record, accumulator, context|
           oclc_number = context.output_hash&.dig('oclc_number_ssim')&.first
-          accumulator << HATHI_ETAS_OVERLAP[oclc_number]
+          accumulator << HATHI_MULTI_OVERLAP&.dig(oclc_number, :ht_bib_key)
           accumulator.compact
         end
       end
+
+      # Extract ht_id
+      def extract_ht_id
+        lambda do |_record, accumulator, context|
+          ht_bib_key = context.output_hash&.dig('ht_bib_key_ssim')&.first
+          if ht_bib_key.nil?
+            oclc_number = context.output_hash&.dig('oclc_number_ssim')&.first
+            accumulator << HATHI_MONO_OVERLAP&.dig(oclc_number, :ht_id)
```

Comment:
Would be nice for this method to be as simple as something like:

```rb
            oclc_number = context.output_hash&.dig('oclc_number_ssim')&.first
            accumulator << HATHI_OVERLAP&.dig(oclc_number, :ht_id)
```

Reply:
```
oclc,item_type,access
1038694,mono,allow
1038694,mono,deny
1038694,multi,allow
1038694,multi,deny
```
because there are records like above, needed to check if there was a `ht_bib_key` here since 1038694 will be in both files.
OR this check could be done while generating the final file to lookup like the final file would have 

I found it to be easier to do this way, but as we discussed, please feel free to give it a try for a combined final file, I might have not think of a sol'n you could come up with.

Reply:
I think we already decided that this record would go to record page so ht_bib_key indexed. Since when dividing up the files I am not checking this logic, I am checking it when indexing ht_id. Sure, it can be checked during massaging the data so the final file would only have 
`1038694,multi,allow` for this record.

Reply:
So does that mean we're trying to make one final file with `oclc`, `ht_id`, `ht_bib_key`, `type` (i.e., `mono` or `multi`) and `access` be nice to deal with? You could use a single easy lookup for any thing and either `ht_id` or `ht_bib_key` would be blank so that record might end up looking like

`1038694,,[the_ht_bib_key_value],multi,allow`

I guess we just can't combine the csvs right now because csvjoin is too costly? Is that the problem? Or is having two files just fine and let's move on! :)

Reply:
I mentioned the one file since I was thinking that is what you wanted to try. I think I'm personally ready to move on from playing around with the data :) But if you come with that one file, I would happily use it :)

Reply:
cvsjoin is said to be costly for big files but I haven't tried with these two files tbh maybe it would be quick since they are smaller now that after removing columns out, It just did not seem necessary since looking up to one file for each field seemed fine. Also then why divide them up in the first place then if you are going with one file approach.

Reply:
yup okay with moving on with two files, all good.